arches/app/media/js/views/components/search/core-search.js

```diff
@@ -0,0 +1,159 @@
+define([
+    'jquery',
+    'underscore',
+    'knockout',
+    'arches',
+    'viewmodels/alert',
+    'views/components/search/base-filter',
+    'templates/views/components/search/core-search.htm',
+], function($, _, ko, arches, AlertViewModel, BaseFilter, coreSearchTemplate) {
+    const componentName = 'core-search';
+    const viewModel = BaseFilter.extend({ 
+        initialize: function(sharedStateObject) {
+            const self = this;
+            sharedStateObject.componentName = componentName;
+            BaseFilter.prototype.initialize.call(this, sharedStateObject);
```

Comment:
I don't think we should think of this as an instance of a `BaseFilter`.  It should rather be it's own thing, or inherit from a new class that mirrors my comments on the `core-search.py` file.